File: bestia-zoneserver/src/test/java/net/bestia/zoneserver/map/path/MapPathfinderFactoryTest.java
package net.bestia.zoneserver.map.path;

import net.bestia.entity.EntityService;
import net.bestia.model.geometry.Point;
import net.bestia.model.map.Map;
import net.bestia.zoneserver.entity.EntitySearchService;
import org.junit.Assert;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.Mock;
import org.mockito.junit.MockitoJUnitRunner;

@RunWith(MockitoJUnitRunner.class)
public class MapPathfinderFactoryTest {
	
	private MapPathfinderFactory factory;
	
	@Mock
	private EntityService entityService;

	@Mock
	private EntitySearchService entitySearchService;
	
	@Mock
	private Map map;
	
	@Before
	public void setup() {
		factory = new MapPathfinderFactory(entityService, entitySearchService);
	}
	
	@Test(expected = NullPointerException.class)
	public void getPathfinder_nullMap_throws() {
		factory.getPathfinder(null);
	}
	
	@Test
	public void getPathfinder_validMap_validPathfinder() {
		Pathfinder<Point> finder = factory.getPathfinder(map);
		Assert.assertNotNull(finder);
	}
}

File: bestia-zoneserver/src/test/kotlin/net/bestia/zoneserver/map/path/MapPathfinderFactoryTest.kt
package net.bestia.zoneserver.map.path

import net.bestia.zoneserver.entity.EntityService
import net.bestia.model.map.BestiaMap
import net.bestia.zoneserver.entity.EntityCollisionService
import org.junit.Assert
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import org.mockito.Mock
import org.mockito.junit.MockitoJUnitRunner

@RunWith(MockitoJUnitRunner::class)
class MapPathfinderFactoryTest {
  private lateinit var factory: MapPathfinderFactory

  @Mock
  private lateinit var entityService: EntityService

  @Mock
  private lateinit var entitiyCollisionService: EntityCollisionService

  @Mock
  private lateinit var map: BestiaMap

  @Before
  fun setup() {
    factory = MapPathfinderFactory(entitiyCollisionService)
  }

  @Test
  fun getPathfinder_validMap_validPathfinder() {
    val finder = factory.getPathfinder(map)
    Assert.assertNotNull(finder)
  }
}
